tools/protobuf-compile/protobuf-compile.go

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+// protobuf-compile is a helper tool for running protoc against all of the
+// .proto files in this repository using specific versions of protoc and
+// protoc-gen-go, to ensure consistent results across all development
+// environments.
+//
+// protoc itself isn't a Go tool, so we need to use a custom strategy to
+// install and run it. The official releases are built only for a subset of
+// platforms that Go can potentially target, so this tool will fail if you
+// are using a platform other than the ones this wrapper tool has explicit
+// support for. In that case you'll need to either run this tool on a supported
+// platform or to recreate what it does manually using a protoc you've built
+// and installed yourself.
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/hashicorp/go-getter"
+)
+
+const protocVersion = "3.15.6"
```

Comment:
This is copied as-is from hashicorp/terraform, hence the mismatching version here. This repo currently uses v5.29.3.

This Go script relies on being able to download binaries from GitHub, so when the code has `protocVersion = "3.15.6"` is can build a URL to download the binary from this release (https://github.com/protocolbuffers/protobuf/releases/tag/v3.15.6) that includes binaries for different combinations of GOOS and GOARCH. However the v5.29.3 release from GitHub doesn't include those binaries 😢 : https://github.com/protocolbuffers/protobuf/releases/tag/v5.29.3

So options would be either to change the version used in this repo to match what's downloadable from GitHub, or change the script to allow downloading versions that aren't distributed as binaries.